iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -99,19 +99,7 @@ private static async Task<T> ReadResponseMessageAsync<T>(HttpResponseMessage mes
                 return (T)(object)message;
             }
 
-            T entity = await ReadAsAsync<T>(message.Content, token).ConfigureAwait(false);
-
-            // ETag in the header is considered authoritative
-            if (entity is IETagHolder eTagHolder)
```

Comment:
No class still implemented this interface in our IoT hub device client, so this if block was never entered.

Reply:
It's worth remembering that HTTP never supported twin operations, and twin operations are the only ones that use ETags in the IoT hub device client